File: content-solutions/implementation/sim-gravity.js
function solution(grid) {
  const g = grid.map((row) => row.split(""));
  const n = g.length;
  const m = g[0].length;
  for (let c = 0; c < m; c++) {
    let land = n - 1;
    for (let r = n - 1; r >= 0; r--) {
      const ch = g[r][c];
      if (ch === "#") land = r - 1;
      else if (ch !== ".") {
        g[r][c] = ".";
        g[land][c] = ch;
        land--;
      }
    }
  }
  return g.map((row) => row.join(""));
}

File: content-solutions/implementation/sim-gravity.py
def solution(grid):
    g = [list(row) for row in grid]
    n, m = len(g), len(g[0])
    for c in range(m):
        land = n - 1
        for r in range(n - 1, -1, -1):
            ch = g[r][c]
            if ch == "#":
                land = r - 1
            elif ch != ".":
                g[r][c] = "."
                g[land][c] = ch
                land -= 1
    return ["".join(row) for row in g]
